idaes/core/base/property_base.py

```diff
@@ -504,6 +505,47 @@ def report(self, index=(0), true_state=False, dof=False, ostream=None, prefix=""
 
         ostream.write("\n" + "=" * max_str_length + "\n")
 
+    def build_port(self, target_block, port_name, doc=None, subset=None):
+        """
+        Constructs a Port based on this StateBlock attached to the target block.
+
+        Args:
+            target_block - block to which Port should be attached
+            port_name - name to use for Port
+            doc - doc string or Prot object
+            subset - slicer representing a subset of indices of StateBlock which
+                should be included in Port
```

Comment:
Should this be called something else? `subset` seems like a confusing name. From this name, I would expect this to be some set "along which" we slice the state block. Instead, it is the slice itself.

Reply:
Do you have a suggestion for an alternative? I can't think of anything better off the top of my head,

Reply:
`slicer`, `slice_`, or even just `slice` come to mind. We could even update this to accept the slice index, e.g. `(slice(None), 0.0)`, then create the slice from this scalar/tuple index (the argument could be called `slice_index` in this case).